backend/src/main/java/com/karankumar/bookproject/backend/model/Book.java

```diff
@@ -244,6 +246,20 @@ public BookBuilder edition(Integer edition) {
         }
     }
 
+    public void addPublisher(@NonNull Publisher publisher) {
+        Set<Publisher> bookPublishers = this.getPublishers();
+        if (bookPublishers == null) {
+            this.setPublishers(Stream.of(publisher).collect(Collectors.toSet()));
+        } else {
+            bookPublishers.add(publisher);
+            this.setPublishers(bookPublishers);
+        }
+    }
+
+    public void removePublisher(@NonNull Publisher publisher) {
+        this.getPublishers().remove(publisher);
```

Comment:
We should also remove the book from the publisher that we're removing. Pseudocode:

```java
publishers.get(publisher).getBooks().remove(this);
```

Instead of using the getter, we can access the member variable directly:

```java
publishers.remove(publisher);
```